### PR TITLE
Get ws URI directly from server URL

### PR DESCRIPTION
**Pull request by hermanliang**
```java
ParseLiveQueryClient<ParseObject> parseLiveQueryClient = ParseLiveQueryClient.Factory.getClient();
```

Get ws URI directly from server URL
ex.
 - https://example.com:1337/parse/ -> wss://example.com:1337/parse/
 - http://example.com:1337/parse/ -> ws://example.com:1337/parse/

### CODE REVIEW

File: ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java [hunk 1]
```diff
@@ -23,14 +25,37 @@
...
 
     private WebSocketClient webSocketClient;
     private int requestIdCount = 1;
     private boolean userInitiatedDisconnect = false;
 
+    /* package */ ParseLiveQueryClientImpl() {
+        this(null);
+        URL serverUrl = ParseRESTCommand.server;
+        if (serverUrl == null) {
+            throw new RuntimeException("serverUrl is null. "
```
Comment (mmimeault) [inline ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java]: checkInit is already called in the other constructor to do that check.

Comment (mmimeault) [inline ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java]: checkInit is already called in the other constructor to do that check.

File: ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java [hunk 2]
```diff
@@ -23,14 +25,37 @@
...
+                url = url.replaceFirst("https", "wss");
+            } else {
+                url = url.replaceFirst("http", "ws");
+            }
+            try {
+                this.uri = new URI(url);
+            } catch (URISyntaxException e) {
+                e.printStackTrace();
+                throw new RuntimeException(e.getMessage());
+            }
```
Comment (mmimeault) [inline ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java]: You could extract that logic into a static util

File: ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java [hunk 3]
```diff
@@ -23,14 +25,37 @@
     private final String applicationId;
     private final String clientKey;
     private final SparseArray<Subscription<? extends ParseObject>> subscriptions = new SparseArray<>();
-    private final URI uri;
+    private URI uri;
```
Comment (mmimeault) [inline ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java]: You should keep it final, and only pass the result WSS URL when it's parsed.

Comment (hermanliang) [inline ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java]: fix it
Thanks

File: ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java (hunk 1 above)
Comment (hermanliang) [inline ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java]: fixed

File: ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java (hunk 2 above)
Comment (hermanliang) [inline ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java]: fixed by 1040abb

File: ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java [hunk 4]
```diff
@@ -45,6 +51,25 @@
         this.webSocketClientCallback = getWebSocketClientCallback();
     }
 
+    private static URI getDefaultUri() {
+        URL serverUrl = ParseRESTCommand.server;
+        if (serverUrl != null) {
```
Comment (rogerhu) [inline ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java]: minor change, but i'd return null serverUrl == null and then take this out of an if block

Comment (hermanliang) [inline ParseLiveQuery/src/main/java/com/parse/ParseLiveQueryClientImpl.java]: changed by d50da48
Thanks 😄

**Pull request merged**